docs/Customization/Dsptools-Blocks.rst

```diff
@@ -0,0 +1,105 @@
+.. _dsptools-blocks:
+
+Dsptools Blocks
+===============
+ 
+Another way to create a MMIO peripheral is to use the Dsptools library for Chisel. In this method, a memory interface is created by creating a "chain". This chain consists of a custom module placed inside a ``DspBlock``, which is then sandwiched between a ``ReadQueue`` and ``WriteQueue``. Those queues then act as memory mapped interfaces to the Rocket Chip SoCs. This section will again primarily focus on designing Tilelink-based peripherals. However, through the resources provided in Dsptools, one could also define an AXI4-based peripheral by following similar steps.
```

Comment:
Can you clarify the difference between adding an MMIO peripheral using this method vs. the "usual" method. In other words: is this a "special case" of the general method, or is this an alternative?